speech/system-test/createRecognizer.v2.test.js

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable */
+
+'use strict';
+
+const {expect} = require('chai');
+const {describe, it} = require('mocha');
+const uuid = require('uuid');
+const sinon = require('sinon');
+const speech = require('@google-cloud/speech').v2;
+
+const {createRecognizerV2} = require('../createRecognizer.v2');
+
+let recognizerName, projectId;
+
+describe('Create a speech recognizer (v2)', () => {
+  const stubConsole = function () {
+    sinon.stub(console, 'error');
+    sinon.stub(console, 'log');
+  }
+
+  const restoreConsole = function () {
+    console.log.restore();
+    console.error.restore();
+  }
+
+  before(async ()=>{
+    const client = new speech.SpeechClient();
+    projectId = await client.getProjectId();
+  })
+
+  beforeEach(stubConsole);
+  afterEach(restoreConsole);
+
+  it('should create a speech recognizer', async () => {
+    recognizerName = await createRecognizerV2(`rec-${uuid.v4()}`, projectId);
```

Comment:
Can we construct `recognizerName` directly in the test, instead of getting it from the sample?

(In other words - can we verify the sample correctly creates + formats the recognizer name?)

Reply:
That's not quite how it works. The recognizerName (a resource URI) is returned by the API; the sample just prints it to the console.. We can (re-)compose it here in the test since we know what the recognizerId is.

I've made some changes to make the test clearer.

Reply:
Got it; thanks for addressing this.

In general, reusing API data is OK _if and only if_ that API data cannot determined by the test itself:
> - If a particular property's value is _randomly chosen by the API_, then (re)using **or** loosely validating that value makes sense.
> - If that property's value can be _calculated within the test itself_, then (where possible) we should try to do that.